src/run_pretrain.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2021 Google LLC
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     https://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+
+import tensorflow as tf
+from official.modeling import tf_utils
+
+import modeling
+import input_utils
+import tensor_utils
+
+
+def model_fn_builder(model_config, init_checkpoint, learning_rate,
+                     num_train_steps, num_warmup_steps, use_tpu,
+                     use_one_hot_embeddings, optimizer, poly_power,
+                     start_warmup_step, learning_rate_schedule):
+  """Returns `model_fn` closure for TPUEstimator."""
+
+  def model_fn(features, labels, mode, params):
+    raise NotImplementedError
```

Comment:
I remember you implemented another model_fn_builder somewhere?

Reply:
I don't think we have another model_fn_builder.